Source/Core/VideoBackends/Vulkan/VKTexture.cpp

```diff
@@ -0,0 +1,339 @@
+// Copyright 2009 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <algorithm>
+#include <cstddef>
+#include <cstring>
+
+#include "Common/Align.h"
+#include "Common/Assert.h"
+#include "Common/CommonTypes.h"
+#include "Common/Logging/Log.h"
+#include "Common/MsgHandler.h"
+
+#include "VideoBackends/Vulkan/CommandBufferManager.h"
+#include "VideoBackends/Vulkan/FramebufferManager.h"
+#include "VideoBackends/Vulkan/StagingTexture2D.h"
+#include "VideoBackends/Vulkan/StateTracker.h"
+#include "VideoBackends/Vulkan/Texture2D.h"
+#include "VideoBackends/Vulkan/Util.h"
+#include "VideoBackends/Vulkan/VKTexture.h"
+#include "VideoBackends/Vulkan/VulkanContext.h"
+
+#include "VideoCommon/ImageWrite.h"
+#include "VideoCommon/TextureConfig.h"
+
+namespace Vulkan
+{
+VKTexture::VKTexture(const TextureConfig& tex_config, std::unique_ptr<Texture2D> texture,
+                     VkFramebuffer framebuffer)
+    : AbstractTexture(tex_config), m_framebuffer(framebuffer), m_texture(std::move(texture))
+{
+}
+
+VKTexture* VKTexture::Create(const TextureConfig& tex_config)
```

Comment:
Seems like freebsd doesn't like that, gives me the same error that Intellisense gave me on Windows. 

`no viable conversion from returned value of type 'unique_ptr<Vulkan::VKTexture, default_delete<Vulkan::VKTexture>>' to function return type 'unique_ptr<AbstractTexture, default_delete<AbstractTexture>>'`

Regardless, I have a workaround.